libbeat/cfgfile/reload.go

```diff
@@ -131,13 +135,15 @@ func (rl *Reloader) Check(runnerFactory RunnerFactory) error {
 
 	debugf("Number of module configs found: %v", len(configs))
 
+	debug.PrintStack()
+	fmt.Printf("CONFIGS %v\n", configs)
 	// Initialize modules
 	for _, c := range configs {
 		// Only add configs to startList which are enabled
 		if !c.Config.Enabled() {
 			continue
 		}
-		_, err := runnerFactory.Create(rl.pipeline, c.Config, c.Meta)
+		err := runnerFactory.CheckConfig(c.Config)
```

Comment:
Does this potentially break other Beats using reload?

Reply:
Sadly, looking at the last CI run I fear the answer is yes.

However, the current behavior breaks `heartbeat`. The problem is that there's no corresponding stop called so if `Create` leaks any resources they'll never be recovered.

I've removed the old code and manually invoked `Stop()`. I do think we should actually use the `CheckConfig` interface instead of doing a `Create` then `Stop`, but for the scope of the PR this is more appropriate.

I will open a separate PR with that refactor.